### PR TITLE
ListMultipartUploads, ListObjectParts return empty response

### PR DESCRIPTION
**Pull request by krisis**
## Description

This change makes ListMultipartUploads and ListObjectParts return empty response. It _doesn't_ change the backend structure.
## Motivation and Context

See #4564 
## How Has This Been Tested?

Using `mc ls -I [-r]` and existing unit tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by krishnasrinivas — APPROVED** [review]

**Comment by krishnasrinivas** [comment]
@krisis one more thing we need to take care of is cleaning up files older than 2 weeks just like gcs.

**Comment by krisis** [comment]
> @krisis one more thing we need to take care of is cleaning up files older than 2 weeks just like gcs.

Is it required as part of this PR?

**Review by fwessels — APPROVED** [review]
LGTM

**Review by harshavardhana — COMMENTED** [review]
Need to know more about this, why are we removing this code without migrating to `.minio.sys` format? is it expected to be removed in this manner without the backend change?

**Comment by krisis** [comment]
> Need to know more about this, why are we removing this code without migrating to .minio.sys format? is it expected to be removed in this manner without the backend change?

@harshavardhana, we decided to do this change before (and independent of) FS backend format changes like possibly using extended attributes for storing user-metadata and bucket policy and the multipart backend to avoid relying for fcntl(3) locks for coordinating among many minio servers in a shared mode.

**Comment by harshavardhana** [comment]
> @harshavardhana, we decided to do this change before (and independent of) FS backend format changes like possibly using extended attributes for storing user-metadata and bucket policy and the multipart backend to avoid relying for fcntl(3) locks for coordinating among many minio servers in a shared mode.

Okay understood now.

**Comment by krisis** [comment]
@krishnasrinivas I have added the periodic cleanup routine similar to GCS gateway as well.

**Comment by deekoder** [comment]
@krishnasrinivas  @fwessels  since the pr is updated since approval reset reviewers.

File: cmd/fs-v1-multipart.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ import (
 	"github.com/minio/sha256-simd"
 )
 
+const (
+	fsMultipartExpiry        = time.Hour * 24 * 14
```
Comment (fwessels) [inline cmd/fs-v1-multipart.go:35]: Maybe use time.Day * 14 ?

File: cmd/fs-v1-multipart.go [hunk 2]
```diff
@@ -978,3 +936,58 @@ func (fs fsObjects) AbortMultipartUpload(bucket, object, uploadID string) error
 
 	return nil
 }
+
+// Remove multipart uploads left unattended in a given bucket older than `fsMultipartExpiry`
+func (fs fsObjects) cleanupStaleMultipartUpload(bucket string) error {
+	var (
+		err error
```
Comment (fwessels) [inline cmd/fs-v1-multipart.go]: You could name the return variable to avoid defining the variable explicitly

File: cmd/fs-v1-multipart.go [hunk 3]
```diff
@@ -978,3 +936,58 @@ func (fs fsObjects) AbortMultipartUpload(bucket, object, uploadID string) error
...
+			return
+		}
+
+		for _, bucketInfo := range bucketInfos {
+			fs.cleanupStaleMultipartUpload(bucketInfo.Name)
+		}
+
+		// Schedule for the next multipart backend cleanup
+		time.Sleep(fsMultipartCleanupPeriod)
+	}
```
Comment (fwessels) [inline cmd/fs-v1-multipart.go:989]: Do we need to guard this thread from crashing? (Or restart it when it crashes?)

**Review by fwessels — APPROVED** [review]
Few minor comments, for the rest LGTM

File: cmd/fs-v1-multipart.go (hunk 1 above)
Comment (krisis) [reply cmd/fs-v1-multipart.go:35]: @fwessels time.Day is not available. I (too) wish it were available.

File: cmd/fs-v1-multipart.go (hunk 2 above)
Comment (krisis) [reply cmd/fs-v1-multipart.go]: Will do.

File: cmd/fs-v1-multipart.go (hunk 3 above)
Comment (krisis) [reply cmd/fs-v1-multipart.go:989]: For now we can keep it simple and not restart on crash. This routine only frees up disk space consumed by 'stale' or 'forgotten' incomplete uploads.

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Making it a point to test this properly with Cloud Berry Client before the release.

**Pull request merged**